src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -277,15 +277,28 @@ public void LogAnalyzerCountSummary()
             ResetDiagnosticLogAggregator();
         }
 
-        private void ResetDiagnosticLogAggregator()
+        public static async Task<IEnumerable<DiagnosticData>> GetDiagnosticsAsync(
```

Comment:
this is actual new code. all other change in the engine is just making existing method static

Reply:
Is this method only invoked from `DefaultDiagnosticIncrementalAnalyzer`? It seems to be the case as we are defaulting lot of things such as `includeSuppressedDiagnostics: false, logAggregatorOpt: null`. If so, would it make sense to actually move this method to `DefaultDiagnosticAnalyzerService`?

Reply:
Seems even `spanOpt: null`, so we should either move the method to default analyzer service or make this parameters such that the callers in `DefaultDiagnosticIncrementalAnalyzer` pass in default values as arguments.

Reply:
if you are okay with diff, I can move all these static methods to its own static helper types or move in to analyzerhelper type. I didn't just because that will make review harder. since you already looked, I am fine to move all these to AnalyzerHelper.

Reply:
tagging @mavasani

Reply:
It's less about having a diff, and more about living on the right objects.  Being static methods floating around somewhere doesn't make a lot of sense to me.  It's unclera why we wouldn't just have these be normal instance methods on the right diagnostic analyzer subclass.